src/kompress/__init__.py

```diff
@@ -157,6 +141,17 @@ class CPath(BasePath):
     _accessor.open has to return file descriptor, doesn't work for compressed stuff.
     """
 
+    def __new__(cls, *args, **kwargs):
+        path = Path(*args)
+        if path.name.endswith(Ext.zip):
```

Comment:
@seanbreckenridge not sure if you see any problem doing that? seems to work well in tests  (there is some 3.8 CI issue I'll fix), will do more trialing with HPI too

Reply:
seems fine to me, yep